### PR TITLE
WAITP-1221 & WAITP-1226: Map overlays

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Move map components that are only used in from ui-map-components to Lesmis
- Split out Marker layer from ui-map-components into the lesmis, web-frontend and tupaia-web since they are all a bit different
- Add PolygonNavigationLayer

There is still an issue with loading the default map overlays so I will look at that in another PR.

---

### Screenshots:
image

image

image

image

### CODE REVIEW

File: packages/tupaia-web/src/AppProviders.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ const queryClient = new QueryClient({
       staleTime: 1000 * 60 * 5, // 5 minutes
       refetchOnWindowFocus: false,
       retry: 1,
+      keepPreviousData: true,
```
Comment (tcaiger) [inline packages/tupaia-web/src/AppProviders.tsx:18]: This means that once data is loaded for a query it keeps that data until the next query is done and stops any flickering that would occur.

File: packages/tupaia-web/src/features/Map/MapOverlays/DataVisualsLayer.tsx [hunk 2]
```diff
@@ -17,14 +25,22 @@ import {
 import { useMapOverlayReport } from '../utils';
 import { EntityCode } from '../../../types';
 import { processMeasureData } from './processMeasureData';
+import styled from 'styled-components';
```
Comment (acdunham) [inline packages/tupaia-web/src/features/Map/MapOverlays/DataVisualsLayer.tsx:28]: This is picky, but just need to reorder the imports here

File: packages/tupaia-web/src/features/Map/MapOverlays/DataVisualsLayer.tsx [hunk 3]
```diff
@@ -87,12 +102,51 @@ export const MarkerLayer = ({ hiddenValues }: { hiddenValues: LegendProps['hidde
     hiddenValues,
   });
 
+  if (!processedMeasureData || !mapOverlayData.serieses) {
+    return null;
+  }
+
+  const serieses = mapOverlayData.serieses;
```
Comment (acdunham) [inline packages/tupaia-web/src/features/Map/MapOverlays/DataVisualsLayer.tsx:109]:
```suggestion
  const { serieses } = mapOverlayData;
```

File: packages/tupaia-web/src/features/Map/MapOverlays/PolygonNavigationLayer.tsx [hunk 4]
```diff
@@ -95,6 +85,16 @@ export const PolygonLayer = () => {
           />
         </>
       )}
+      {childEntities &&
+        childEntities.length > 0 &&
```
Comment (acdunham) [inline packages/tupaia-web/src/features/Map/MapOverlays/PolygonNavigationLayer.tsx:89]:
```suggestion
        {childEntities?.length > 0 &&
```

File: packages/tupaia-web/src/features/Map/MapOverlays/PolygonNavigationLayer.tsx [hunk 5]
```diff
@@ -95,6 +85,16 @@ export const PolygonLayer = () => {
           />
         </>
       )}
+      {childEntities &&
```
Comment (acdunham) [inline packages/tupaia-web/src/features/Map/MapOverlays/PolygonNavigationLayer.tsx:88]:
```suggestion
      
```

File: packages/tupaia-web/src/features/Map/MapOverlays/processMeasureData.ts [hunk 6]
```diff
@@ -49,4 +50,14 @@ export const processMeasureData = ({
...
   });
+
+  // for radius overlay sort desc radius to place smaller circles over larger circles
+  if (serieses.some(l => l.type === MEASURE_TYPE_RADIUS)) {
+    entityMeasureData.sort((a, b) => Number(b.radius) - Number(a.radius));
+  }
+
+  // Filter hidden and invalid values and sort measure data
+  return entityMeasureData
+    .filter(({ coordinates, region }) => region || (coordinates && coordinates?.length === 2))
```
Comment (acdunham) [inline packages/tupaia-web/src/features/Map/MapOverlays/processMeasureData.ts:61]:
```suggestion
    .filter(({ coordinates, region, isHidden }) => (region || (coordinates && coordinates?.length === 2)) && !isHidden)
```

File: packages/web-frontend/src/containers/Map/DataVisualsLayer/MarkerLayer.js [hunk 7]
```diff
@@ -5,11 +5,66 @@
...
+  MeasurePopup,
+  MEASURE_TYPE_RADIUS,
+  LayerGroup,
+} from '@tupaia/ui-map-components';
 import { selectMeasureOptions } from '../../../selectors';
 
+// Filter hidden and invalid values and sort measure data
+const processData = (measureData, serieses) => {
+  const data = measureData
+    .filter(({ coordinates, region }) => region || (coordinates && coordinates?.length === 2))
```
Comment (acdunham) [inline packages/web-frontend/src/containers/Map/DataVisualsLayer/MarkerLayer.js:20]:
```suggestion
    .filter(({ coordinates, region, isHidden }) => (region || (coordinates && coordinates?.length === 2)) && !isHidden)
```

**Review by acdunham — APPROVED** [review]
This is a huge piece of work! I have some super small pedantic suggestions but nothing major. I think it's a good idea to separate things out, and means we can just take what we need, which seems like not that much in the end 🎉

**Pull request merged**